skbase/lookup/tests/test_lookup.py

```diff
@@ -0,0 +1,899 @@
+# -*- coding: utf-8 -*-
+# copyright: skbase developers, BSD-3-Clause License (see LICENSE file)
+"""Tests for skbase lookup functionality.
+
+tests in this module:
+
+    test_is_non_public_module  - tests _is_non_public_module logic
+"""
+import importlib
+import pathlib
+from types import ModuleType
+from typing import List
+
+import pandas as pd
+import pytest
+
+from skbase.base import BaseEstimator, BaseObject
+from skbase.base._base import TagAliaserMixin
+from skbase.lookup import all_objects, get_package_metadata
+from skbase.lookup._lookup import (
+    _determine_module_path,
+    _filter_by_class,
+    _filter_by_tags,
+    _get_return_tags,
+    _import_module,
+    _is_ignored_module,
+    _is_non_public_module,
+    _walk,
+)
+from skbase.mock_package.mock_package import CompositionDummy, NotABaseObject
+from skbase.tests.conftest import FixtureClassParent
+
+__author__: List[str] = ["RNKuhns"]
+__all__: List[str] = []
+
+
+MODULE_METADATA_EXPECTED_KEYS = (
+    "path",
+    "name",
+    "classes",
+    "functions",
+    "__all__",
+    "authors",
+    "is_package",
+    "contains_concrete_class_implementations",
+    "contains_base_classes",
+    "contains_base_objects",
+)
+MOD_NAMES = {
+    "public": ("skbase", "skbase.lookup", "some_module.some_sub_module"),
+    "non_public": (
+        "skbase.lookup._lookup",
+        "some_module._some_non_public_sub_module",
+        "_skbase",
+    ),
+}
+REQUIRED_CLASS_METADATA_KEYS = [
+    "klass",
+    "name",
+    "description",
+    "tags",
+    "is_concreate_implementation",
+    "is_base_class",
+    "is_base_object",
+    "authors",
+    "module_name",
+]
+REQUIRED_FUNCTION_METADATA_KEYS = ["func", "name", "description", "module_name"]
+
+
+@pytest.fixture
```

Comment:
do we need these fixtures if classes are returned?

Fixtures are typically used for python objects that have identity, but classes are not among those (objects would be)

Reply:
Yep, that makes sense. I'll reserve them for things like objects, dicts and the like.